platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -343,6 +372,31 @@ def _read_generic_file(self, filename, len):
             return '0'
 
 
+    def _parse_dmi(self, filename):
+        """
+        Read DMI data chassis data and returns a dictionary of values
+
+        Returns:
+            A dictionary containing the dmi table of the switch chassis info
+        """
+        result = {}
+        try:
+            fileobj = open(filename, "rb")
```

Comment:
maybe can use "with open" ?

Reply:
Fine with me. Was attempting to use the same pattern as in the `read_file()` method above for general consistency.